zellij-client/src/lib.rs

```diff
@@ -213,7 +213,7 @@ pub fn start_client(
                         let send_client_instructions = send_client_instructions.clone();
                         move || {
                             send_client_instructions
-                                .send(ClientInstruction::Exit(ExitReason::Normal))
+                                .send(ClientInstruction::Exit(ExitReason::ForceDetached))
```

Comment:
Does this behave as expected?
I think we should rather be sending `ClientToServerMsg::Action(Action::Detach)` to the server for the actual detach to happen.
(Even the existing logic seems suspicious to me. We should have been sending `ClientToServerMsg::Action(Action::Quit)` for a clean exit. Must have missed changing this part when the refactor happened.)

Reply:
Hey @kunalmohan! In my tests with this code I was able to force close zellij (with interrupt signals) and see the session available with list-sessions - that didn't happen with main. But I'm ok with doing some refactoring.

On a related note, what do you think about changing the behavior for `zellij attach` so instead of an error if there are no detached sessions, it would start a new one. What I'm thinking is to always start zellij with the attach option so I recover some eventually detached session when available or create a new one otherwise. Maybe recovering the previous session  should even be the default.

Reply:
I believe it is being worked on in https://github.com/zellij-org/zellij/issues/541. The idea is to have a --create (-c) flag with the attach command to create a new session if one does not already exist. I'm not sure about making it default. Attaching to an existing session and creating a new one are two separate operations. Attach command having a default behaviour of "attach or create" sounds a bit weird and confusing. IMO having a config option to change the default behaviour of this command can be a good option.

Reply:
But that would involve me knowing the name of a session to use with --create (session-name). My suggestion was more in the line that if a user has a detached session (because it was previously interrupted) when he opens Zellij again he probably wants to continue from where he left, would be a nice UX not having to know that session name. If you have just one session  detached  "recover" it instead of creating new sessions.
So the proposal would be "zellij attach" attaches if there is one session available or creates a new one otherwise.

Reply:
How about `zellij a -c`? The session name is already optional for attach command, so in case of `zellij a -c`:
1. If only single session exists, recover it.
2. If more than one sessions exists, exit with a message (as we do now).
3. If no session exists, create a new one with a random name.

Reply:
Re: Whether `zellij attach` should have a separate `--create` flag or a default behaviour of "recover-or-create". I'm inclined towards the former, but don't have a strong opinion about it. I'm guessing that other folks might also have an opinion about this, so maybe we can move this discussion to discord or a new issue? We can then decide accordingly.

Reply:
Thanks @kunalmohan ! I honestly would be happy with both scenarios (default or not) as long as we have the option to "recover or create" without requiring a name to recover.
Yes, lets continue this on Discord.